batch/batch/worker/worker_api.py

```diff
@@ -26,7 +26,7 @@ def get_cloud_async_fs(self) -> AsyncFS:
         raise NotImplementedError
 
     @abc.abstractmethod
-    def user_credentials(self, credentials: Dict[str, bytes]) -> CloudUserCredentials:
+    def user_credentials(self, credentials: Dict[str, str]) -> CloudUserCredentials:
```

Comment:
I think this was wrong as `bytes`. If it is a JSON object it can't have a `bytes` value because that's not valid in the JSON spec. It needs to be a number, bool, str, list, or dict. Example:

```
>>> json.dumps({'a': b'abc'})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/dgoldste/opt/miniconda3/envs/hail/lib/python3.7/json/__init__.py", line 231, in dumps
    return _default_encoder.encode(obj)
  File "/Users/dgoldste/opt/miniconda3/envs/hail/lib/python3.7/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/Users/dgoldste/opt/miniconda3/envs/hail/lib/python3.7/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/Users/dgoldste/opt/miniconda3/envs/hail/lib/python3.7/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type bytes is not JSON serializable
```

Reply:
In the azure credentials we also directory return a value from this dict for username and password, which I presume would be `str`.